torch/fx/experimental/symbolic_shapes.py

```diff
@@ -411,21 +426,19 @@ def guard_scalar(a):
 
 
 @record_shapeenv_event()
-def _constrain_symbol_range(shape_env, s: sympy.Symbol, compiler_min: int, compiler_max: int, runtime_min: int, runtime_max: int):
-    log.debug("_constrain_symbol_range %s [%s, %s] [%s, %s]", s, compiler_min, compiler_max, runtime_min, runtime_max)
+def _constrain_symbol_range(shape_env, s: sympy.Symbol, compiler_min: int, compiler_max: int):
+    changed = False
     if r := shape_env.var_to_range.get(s, None):
-        shape_env.var_to_range[s] = ValueRanges(
+        ret = shape_env.var_to_range[s] = ValueRanges(
             builtins.max(r.lower, compiler_min), builtins.min(r.upper, compiler_max)
         )
```

Comment:
We should have a builtin ValueRanges method for this. intersection ?

Reply:
We do, can just switch this to intersection lol.

Reply:
Right now intersection will throw on non-bools but we can extend it

Reply:
it's fine

```
        assert self.is_bool == other.is_bool, (self, other)
```